.eslintrc.js

```diff
@@ -1,45 +1,23 @@
 module.exports = {
-  env: {
-    browser: true,
-    es6: true,
-  },
   extends: [
-    'standard',
-    'standard-react',
-    'prettier/react',
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'prettier/@typescript-eslint',
     'plugin:prettier/recommended',
+    'react-app',
   ],
-  parser: 'babel-eslint',
+  plugins: ['@typescript-eslint', 'prettier'],
   parserOptions: {
-    ecmaFeatures: {
-      experimentalObjectRestSpread: true,
-      jsx: true,
-    },
-    sourceType: 'module',
-  },
-  plugins: ['prettier', 'react', 'react-hooks'],
-  rules: {
-    'react/prop-types': 'warn',
-    'react-hooks/rules-of-hooks': 'error',
-    'react-hooks/exhaustive-deps': 'error',
-    'import/no-unresolved': [
-      'error',
-      { ignore: ['^react(-dom)?$', '^styled-components$'] },
-    ],
-    'promise/no-nesting': ['off'],
-    'linebreak-style': ['error', 'unix'],
-    curly: 'error',
+    jsx: true,
+    project: './tsconfig.json',
   },
   settings: {
     react: {
-      pragma: 'React',
-      version: '16.6',
+      version: '999.999.999',
     },
   },
-  overrides: [
-    {
-      files: ['**/*.test.js'],
-      env: { jest: true },
-    },
-  ],
+  rules: {
+    '@typescript-eslint/no-non-null-assertion': 'off',
```

Comment:
I didn’t know this rule existed! I would find it super useful to have in our projects. Any reason why it is disabled here?

Reply:
I think this originated from standardising config between a couple of other typescript PRs. I agree it's useful to have enabled.

Non null assertion can be helpful in certain react cases (e.g. context) where you know you're safe with an initial `undefined`, but for that we can cast which might be better actually. I've removed 👍